dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/EventProgramEnrollmentService.java

```diff
@@ -36,12 +36,15 @@
  */
 public interface EventProgramEnrollmentService {
 
-  /** Returns a list of enrollments in the given program. */
-  List<Enrollment> getEnrollments(Program program);
-
   /**
    * Returns the list of enrollments in the given program, only if the program is of type {@link
    * ProgramType#WITHOUT_REGISTRATION}
    */
+  List<Enrollment> getEnrollments(Program program);
+
+  /**
+   * Returns the list of enrollments in the given program and status, only if the program is of type
+   * {@link ProgramType#WITHOUT_REGISTRATION}
+   */
   List<Enrollment> getEnrollments(Program program, EnrollmentStatus enrollmentStatus);
```

Comment:
I am wondering if the actual method names could/should be changed to make it self-explanatory? 
 If the method only returns enrollments for event programs, the name "getEnrollments(..)" is misleading right? The javadoc might be overlooked by a consumer. 

The interface is EventProgramEnrollmentService though. So that may suffice. What are your thoughts?

Reply:
the method name reflecting what it's actually doing sounds like a good improvement to me 👍

Reply:
In my mind, naming the interface `EventProgramEnrollment` was meant to indicate that any method under it would deal exclusively with event programs. I didn't want to make the name redundant by calling each method `getEventProgramEnrollments`. I added the Javadoc in case someone is using it and doesn't know the difference between event and tracker programs.

Anyway, this service should be temporary, since the long-term idea is to separate tracker programs from event programs so that they will be different models. Once we achieve that, this service will no longer be needed.

Reply:
Sounds good to me. 👍